# US - Baby Names

### Introduction:

We are going to use a subset of [US Baby Names](https://www.kaggle.com/kaggle/us-baby-names) from Kaggle.  
In the file it will be names from 2004 until 2014


### Step 1. Import the necessary libraries

In [1]:
import pandas as pd
import numpy

### Step 2. Import the dataset from this US_Baby_Names_right.csv 

### Step 3. Assign it to a variable called baby_names.

In [25]:
baby_names = pd.read_csv('US_Baby_Names_right.csv')

### Step 4. See the first 10 entries

In [26]:
baby_names.head(10)

Unnamed: 0     Id      Name  Year Gender State  Count
0       11349  11350      Emma  2004      F    AK     62
1       11350  11351   Madison  2004      F    AK     48
2       11351  11352    Hannah  2004      F    AK     46
3       11352  11353     Grace  2004      F    AK     44
4       11353  11354     Emily  2004      F    AK     41
5       11354  11355   Abigail  2004      F    AK     37
6       11355  11356    Olivia  2004      F    AK     33
7       11356  11357  Isabella  2004      F    AK     30
8       11357  11358    Alyssa  2004      F    AK     29
9       11358  11359    Sophia  2004      F    AK     28

### Step 5. Delete the column 'Unnamed: 0' and 'Id'

In [27]:
del baby_names['Unnamed: 0']
del baby_names['Id']

In [28]:
baby_names.head(10)

Name  Year Gender State  Count
0      Emma  2004      F    AK     62
1   Madison  2004      F    AK     48
2    Hannah  2004      F    AK     46
3     Grace  2004      F    AK     44
4     Emily  2004      F    AK     41
5   Abigail  2004      F    AK     37
6    Olivia  2004      F    AK     33
7  Isabella  2004      F    AK     30
8    Alyssa  2004      F    AK     29
9    Sophia  2004      F    AK     28

### Step 6. Is there more male or female names in the dataset?

In [36]:
baby_names.groupby('Gender').Count.describe()
#drinks.groupby('continent').wine_servings.describe()

count       mean         std  min  25%   50%   75%     max
Gender                                                               
F       558846.0  29.310925   75.962992  5.0  6.0  10.0  23.0  3634.0
M       457549.0  41.615650  118.074308  5.0  7.0  12.0  29.0  4167.0

Vemos que hay mas niñas que niños.

### Step 7. Group the dataset by name and assign to names

In [39]:
names = baby_names.groupby('Name').sum()

### Step 8. How many different names exist in the dataset?

In [41]:
len(names)

17632

Vemos que hay 17632 nombres diferentes.

### Step 9. What is the name with most occurrences?

In [46]:
names.Count.idxmax()

'Jacob'

El nombre que más se repite es Jacob.

### Step 10. How many different names have the least occurrences?

In [51]:
len(names[names.Count == names.Count.min()])

2578

Vemos que hay 2578.

### Step 11. What is the median name occurrence?

In [52]:
names.Count.median()

49.0

### Step 12. What is the standard deviation of names?

In [53]:
names.Count.std()

11006.069467891111

### Step 13. Get a summary with the mean, min, max, std and quartiles.

In [54]:
names.Count.describe()

count     17632.000000
mean       2008.932169
std       11006.069468
min           5.000000
25%          11.000000
50%          49.000000
75%         337.000000
max      242874.000000
Name: Count, dtype: float64